src/noit_message_decoder.c

```diff
@@ -334,17 +397,15 @@ noit_metric_tags_parse_one(const char *tagnm, size_t tagnmlen,
 }
 
 static int
-noit_metric_tags_compare(const void *v_l, const void *v_r) {
-  const noit_metric_tag_t *l = (noit_metric_tag_t *) v_l;
-  const noit_metric_tag_t *r = (noit_metric_tag_t *) v_r;
-  size_t memcmp_len = l->total_size < r->total_size ? l->total_size : r->total_size;
-  int cmp_rslt = memcmp(l->tag, r->tag, memcmp_len);
-  if(cmp_rslt != 0) return cmp_rslt;
-  if(l->total_size < r->total_size) return -1;
-  if(l->total_size > r->total_size) return 1;
-  return 0;
+tag_canonical_size(noit_metric_tag_t *tag) {
+  int len;
+  char dbuff[NOIT_TAG_MAX_PAIR_LEN], ebuff[NOIT_TAG_MAX_PAIR_LEN];
```

Comment:
Don't need `ebuff` here since this is just throwaway decode/encode for sizing reasons.